geth/common/utils.go

```diff
@@ -65,26 +72,52 @@ func ToAddress(accountAddress string) *common.Address {
 	return &to.Address
 }
 
-// ImportTestAccount checks if test account exists in keystore, and if not
-// tries to import it (from static resources, see "static/keys" folder)
-func ImportTestAccount(keystoreDir, accountFile string) error {
-	// make sure that keystore folder exists
-	if _, err := os.Stat(keystoreDir); os.IsNotExist(err) {
-		os.MkdirAll(keystoreDir, os.ModePerm) // nolint: errcheck
+// RestoreFile checks if decrypted file exists in dir, and if not
+// tries to restore and decrypt it (from static resources, see "static/keys" folder)
+func RestoreFile(dir, file string) error {
+	text, err := ReadEncryptedFile(dir, file)
+	if err != nil {
+		return err
+	}
+	if len(text) == 0 {
+		return nil
 	}
 
-	dst := filepath.Join(keystoreDir, accountFile)
-	if _, err := os.Stat(dst); os.IsNotExist(err) {
-		err = ioutil.WriteFile(dst, static.MustAsset("keys/"+accountFile), 0644)
-		if err != nil {
-			log.Warn("cannot copy test account PK", "error", err)
-			return err
-		}
+	dst := filepath.Join(dir, file)
+	err = ioutil.WriteFile(dst, text, 0644)
+	if err != nil {
+		log.Warn("cannot restore file", "error", err)
```

Comment:
It seems to me that logging must be handled by the calling function if necessary, it other words returning error should be enough, no need to log it here.